lib/srv/sessiontracker.go

```diff
@@ -181,6 +181,26 @@ func (s *SessionTracker) WaitForStateUpdate(initialState types.SessionState) typ
 	}
 }
 
+// WaitOnState waits until the desired state is reached. The context may be used for cancellation
+// but the condvar must be manually signaled for this to take effect.
```

Comment:
Hmm, sorry for sending you down a few different paths, but I don't love this either, as it seems difficult to actually cancel the wait correctly.

Condvars and channels just don't play well together unfortunately.

Removing the use of condvars in the session tracker is probably a big lift. Given that this is only used in a test, maybe a good compromise for now would just be to unexport this so it's less likely to be used in other parts of the code base?

Reply:
Actually, have a look again now. We can do this within the function.